### PR TITLE
fix: missing properties in group type definition

### PR DESCRIPTION
**Pull request by gekkedev**
Corrected the `GroupMetadata` interface based on two example groups I retrieved using `client.listChats({ onlyGroups: true })`.

- it has a few TODOs but I thought generic types are better than none
- added a parameter correction originally proposed here: https://github.com/wppconnect-team/wppconnect/pull/2248#discussion_r1660307272

### CODE REVIEW

**Comment by icleitoncosta** [comment]
Thanks, please change comma for dot and comma

File: src/api/model/group-metadata.ts [hunk 1]
```diff
@@ -29,13 +35,32 @@ export interface GroupMetadata {
...
   ephemeralDuration: number;
+  membershipApprovalMode: boolean;
+  memberAddMode: 'admin_add' | string; //TODO: complete the union type and then remove `strong`
+  reportToAdminMode: boolean;
+  /** current amount of members including admins */
   size: number;
   support: boolean;
   suspended: boolean;
   terminated: boolean;
+  uniqueShortNameMap: Object, //TODO: type is too generic
```
Comment (gekkedev) [inline src/api/model/group-metadata.ts:46]:
```suggestion
  uniqueShortNameMap: Object; //TODO: type is too generic
```

File: src/api/model/group-metadata.ts [hunk 2]
```diff
@@ -29,13 +35,32 @@ export interface GroupMetadata {
...
   isParentGroup: boolean;
+  isParentGroupClosed: boolean;
   defaultSubgroup: boolean;
+  generalSubgroup: boolean;
+  generalChatAutoAddDisabled: boolean;
+  allowNonAdminSubGroupCreation: boolean;
+  lastActivityTimestamp: number,
+  lastSeenActivityTimestamp: number,
+  incognito: boolean,
+  hasCapi: boolean,
```
Comment (gekkedev) [inline src/api/model/group-metadata.ts:57]:
```suggestion
  hasCapi: boolean;
```

File: src/api/model/group-metadata.ts [hunk 3]
```diff
@@ -29,13 +35,32 @@ export interface GroupMetadata {
...
+  isLidAddressingMode: boolean,
   isParentGroup: boolean;
+  isParentGroupClosed: boolean;
   defaultSubgroup: boolean;
+  generalSubgroup: boolean;
+  generalChatAutoAddDisabled: boolean;
+  allowNonAdminSubGroupCreation: boolean;
+  lastActivityTimestamp: number,
+  lastSeenActivityTimestamp: number,
+  incognito: boolean,
```
Comment (gekkedev) [inline src/api/model/group-metadata.ts:56]:
```suggestion
  incognito: boolean;
```

File: src/api/model/group-metadata.ts [hunk 4]
```diff
@@ -29,13 +35,32 @@ export interface GroupMetadata {
...
+  uniqueShortNameMap: Object, //TODO: type is too generic
+  isLidAddressingMode: boolean,
   isParentGroup: boolean;
+  isParentGroupClosed: boolean;
   defaultSubgroup: boolean;
+  generalSubgroup: boolean;
+  generalChatAutoAddDisabled: boolean;
+  allowNonAdminSubGroupCreation: boolean;
+  lastActivityTimestamp: number,
+  lastSeenActivityTimestamp: number,
```
Comment (gekkedev) [inline src/api/model/group-metadata.ts:55]:
```suggestion
  lastSeenActivityTimestamp: number;
```

File: src/api/model/group-metadata.ts [hunk 5]
```diff
@@ -29,13 +35,32 @@ export interface GroupMetadata {
...
   terminated: boolean;
+  uniqueShortNameMap: Object; //TODO: type is too generic
+  isLidAddressingMode: boolean;
   isParentGroup: boolean;
+  isParentGroupClosed: boolean;
   defaultSubgroup: boolean;
+  generalSubgroup: boolean;
+  generalChatAutoAddDisabled: boolean;
+  allowNonAdminSubGroupCreation: boolean;
+  lastActivityTimestamp: number,
```
Comment (gekkedev) [inline src/api/model/group-metadata.ts:54]:
```suggestion
  lastActivityTimestamp: number;
```

**Comment by gekkedev** [comment]
done @icleitoncosta :-)

**Comment by gekkedev** [comment]
I want to amend the return type of `getGroupInfoFromInviteLink`, where can I find the type definition? In the repo for wa-js I found 3 similar type definitions but they had some differences to return type the latest version (so I assume those are not the ones I was looking for).

**Comment by gekkedev** [comment]
> I also want to amend the return type of `getGroupInfoFromInviteLink`, where can I find the type definition? In the repo for wa-js I found 3 similar type definitions but they all had some differences to the return types shipped in the latest version (so I assume those are not the ones I was looking for).

@icleitoncosta could you take a look at how I can change this? Got more info to add but I'm a little lost.

File: src/api/layers/sender.layer.ts [hunk 6]
```diff
@@ -234,7 +234,7 @@ export class SenderLayer extends ListenerLayer {
    * Sends image message
    * @category Chat
    * @param to Chat id
-   * @param base64 File path, http link or base64Encoded
+   * @param base64  A base64-encoded data URI (with mime type)
```
Comment (gekkedev) [inline src/api/layers/sender.layer.ts:237]: duplicate space
```suggestion
   * @param base64 A base64-encoded data URI (with mime type)
```

**Comment by gekkedev** [comment]
@icleitoncosta I adjusted the commit names

**Pull request merged**